### PR TITLE
[WIP] Add integration test for HTTP/2 inbound endpoint.

### PR DESCRIPTION
**Pull request by thuva9872**
## Purpose
HTTP/2 support is added in Synapse as inbound endpoints using Netty. Integration tests are done to ensure the HTTP/2 support.

## Goals
Integration tests are added to test the HTTP/2 inbound endpoints.

## Approach
Client to ESB HTTP/2 request-response support is checked using an HTTP/2 client.
ESB to Backend HTTP/2 request-response support is checked with the help of Jetty servers.

## Related PRs
https://github.com/wso2/wso2-synapse/pull/1949

## Test environment
JDK 11

### CODE REVIEW

File: integration/mediation-tests/tests-transport-2/pom.xml [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <source>8</source>
+                    <target>8</target>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
+                <version>2.4</version>
```
Comment (Nirothipan) [inline integration/mediation-tests/tests-transport-2/pom.xml:49]: version should be inherited from parent pom

File: integration/mediation-tests/tests-transport-2/pom.xml [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+        </plugins>
+    </build>
+
+    <properties>
+        <maven.compiler.source>8</maven.compiler.source>
+        <maven.compiler.target>8</maven.compiler.target>
+        <jetty.version>11.0.9</jetty.version>
+        <io.netty.version>4.1.77.Final</io.netty.version>
+        <io.netty.tcnative.version>2.0.52.Final</io.netty.tcnative.version>
+    </properties>
```
Comment (Nirothipan) [inline integration/mediation-tests/tests-transport-2/pom.xml:228]: these should go to parent pom dependency management

File: integration/mediation-tests/tests-transport-2/src/test/java/org/wso2/carbon/esb/http2/test/BackendServer.java [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+        this.serverConnector =
+                httpWsConnectorFactory.createServerConnector(new ServerBootstrapConfiguration(new HashMap<>()),
+                        listenerConfiguration);
+        ServerConnectorFuture serverConnectorFuture = serverConnector.start();
+        serverConnectorFuture.setHttpConnectorListener(new DefaultHttpConnectorListener());
+        try {
+            serverConnectorFuture.sync();
+        } catch (InterruptedException e) {
+            log.error("Error while starting the server");
```
Comment (Nirothipan) [inline integration/mediation-tests/tests-transport-2/src/test/java/org/wso2/carbon/esb/http2/test/BackendServer.java:73]:
```suggestion
            log.error("Error while starting the server", e);
```

File: integration/mediation-tests/tests-transport-2/src/test/java/org/wso2/carbon/esb/http2/test/ClientToESBHttp2Test.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    @Test(description = "Test HTTP/2 request-response between the client and ESB")
+    public void testClientToESBHttp2Request() throws IOException {
+
+        OkHttpClient client =
+                new OkHttpClient.Builder().protocols(Collections.singletonList(Protocol.H2_PRIOR_KNOWLEDGE)).build();
+        Request request = new Request.Builder().url("http://localhost:5055/noBackendCall").get().build();
+        Response response = client.newCall(request).execute();
+        Assert.assertEquals(response.protocol(), Protocol.H2_PRIOR_KNOWLEDGE);
```
Comment (Nirothipan) [inline integration/mediation-tests/tests-transport-2/src/test/java/org/wso2/carbon/esb/http2/test/ClientToESBHttp2Test.java:50]: add the assert message

File: integration/mediation-tests/tests-transport-2/src/test/java/org/wso2/carbon/esb/http2/test/ClientToESBSecureHttp2Test.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        super.init();
+    }
+
+    @Test(description = "Test the secure HTTP/2 (HTTPS) request-response between the client and ESB")
+    public void testSecureHttp2RequestClientToESB() throws IOException {
+
+        OkHttpClient client = new OkHttpClient.Builder().build();
+        Request request = new Request.Builder().url("https://localhost:5100/noBackendCall").get().build();
+        Response response = client.newCall(request).execute();
+        Assert.assertEquals(response.protocol(), Protocol.HTTP_2);
```
Comment (Nirothipan) [inline integration/mediation-tests/tests-transport-2/src/test/java/org/wso2/carbon/esb/http2/test/ClientToESBSecureHttp2Test.java:48]: add assert msg

File: integration/mediation-tests/tests-transport-2/src/test/resources/testng.xml [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        <listener class-name="org.wso2.carbon.automation.engine.testlisteners.TestSuiteListener"/>
+        <listener class-name="org.wso2.carbon.automation.engine.testlisteners.TestTransformerListener"/>
+    </listeners>
+    <test name="HTTP2-Single-Request-Test" preserve-order="true" verbose="2">
+        <packages>
+            <package name="org.wso2.carbon.esb.http2.test"/>
+        </packages>
+    </test>
+</suite>
+
```
Comment (Nirothipan) [inline integration/mediation-tests/tests-transport-2/src/test/resources/testng.xml:36]: remove this extra line

**Comment by thuva9872** [comment]
new PR: https://github.com/wso2/micro-integrator/pull/2711

**Pull request closed without merging**